### PR TITLE
WIP Read arrow

### PR DESCRIPTION
**Pull request by willdealtry**
#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [ ] Have you updated the relevant docstrings, documentation and copyright notice?
 - [ ] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [ ] Are API changes highlighted in the PR description?
 - [ ] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

File: cpp/arcticdb/util/error_code.hpp [hunk 1]
```diff
@@ -80,8 +82,9 @@ inline std::unordered_map<ErrorCategory, const char*> get_error_category_names()
     ERROR_CODE(7000, E_INVALID_USER_ARGUMENT) \
     ERROR_CODE(7001, E_INVALID_DECIMAL_STRING)   \
     ERROR_CODE(7002, E_INVALID_CHAR_IN_SYMBOL) \
-    ERROR_CODE(8000, E_UNRECOGNISED_COLUMN_STATS_VERSION) \
+    ERROR_CODE(8000, E_UNRECOGNISED_COLUMN_STATS_VERSION)          \
     ERROR_CODE(9000, E_DECODE_ERROR)
```
Comment (jjerphan) [inline cpp/arcticdb/util/error_code.hpp:86]: The escape is required for the project compile.
```suggestion
    ERROR_CODE(8000, E_UNRECOGNISED_COLUMN_STATS_VERSION) \
    ERROR_CODE(9000, E_DECODE_ERROR) \
```

File: cpp/arcticdb/column_store/column.hpp [hunk 2]
```diff
@@ -400,6 +400,13 @@ class Column {
 
     void set_allow_sparse(bool value);
 
+    void force_set_row_data(size_t row_id) {
+        last_physical_row_ = row_id;
+        last_logical_row_ = row_id;
+    }
+
+    size_t get_physical_offset(size_t row) const;
+
```
Comment (jjerphan) [inline cpp/arcticdb/column_store/column.hpp:409]: This is duplicated from line 393.
```suggestion
```

File: cpp/arcticdb/arrow/arrow_schema.hpp [hunk 3]
```diff
@@ -0,0 +1,289 @@
+#pragma once
+
+#include <arcticdb/arrow/arrow_c_data_interface.hpp>
+#include <arcticdb/arrow/arrow_wrappers.h>
+#include <arcticdb/arrow/nanoarrow/nanoarrow.h>
+#include <arcticdb/arrow/arrow_error.hpp>
+#include <arcticdb/util/preconditions.hpp>
+
```
Comment (jjerphan) [inline cpp/arcticdb/arrow/arrow_schema.hpp:8]: `std::optional` is unresolved with gcc 12.3.0-3.
```suggestion

#include <optional>
```

**Review by jjerphan — COMMENTED** [review]
Thank you for exploring this, @willdealtry.

Here are a few changes I needed to add to compile the project on my machine.

**Comment by willdealtry** [comment]
This was a PoC, we are waiting for sparrow to implement this properly for us

**Pull request closed without merging**